commands/poll.js

```diff
@@ -0,0 +1,333 @@
+const { SlashCommandBuilder } = require("@discordjs/builders");
+const {
+  ActionRowBuilder,
+  ButtonBuilder,
+  ButtonStyle,
+  ComponentType,
+  EmbedBuilder,
+} = require("discord.js");
+const dotenv = require("dotenv");
+
+dotenv.config();
+
+const MAX_OPTIONS = 10;
+const OPTION_EMOJIS = [
+  "1️⃣",
+  "2️⃣",
+  "3️⃣",
+  "4️⃣",
+  "5️⃣",
+  "6️⃣",
+  "7️⃣",
+  "8️⃣",
+  "9️⃣",
+  "🔟",
+];
+const CLIENT_ID = process.env.DISCORD_CLIENT_ID;
+const DELAY = 3;
+
+const getTopVotedOption = async (message, interaction) => {
+  const initialResults = await Promise.all(
+    message.reactions.cache.map(async (reaction) => {
+      const emojiName = reaction.emoji.name;
+      const reactionUserIds = await reaction.users.fetch();
+
+      // excludes bot
+      const reactionUsernames = await Promise.all(
+        [...reactionUserIds.keys()]
+          .filter((userId) => userId !== CLIENT_ID)
+          .map(async (userId) => {
+            const user = await interaction.client.users.fetch(userId);
+            if (!user.bot) {
+              return user.username;
+            }
+          })
+      );
+
+      const reactionCount = reactionUsernames.length;
```

Comment:
Suggestion: For a specific reaction, we can actually get the number of users who reacted by using [count](https://old.discordjs.dev/#/docs/discord.js/14.9.0/class/MessageReaction?scrollTo=count). So we can probably have `reactionCount` be basically `reaction.count - 1` rather than deriving it from the `length` of `reactionUsernames`.

It still makes sense to filter the list of users who reacted to a specific reaction by excluding the bot, since the bot would react with all reactions first upon creating the poll.

Also, given that you have already filtered out the user ID of the bot, we probably don't need to check if the user is a bot regardless. I mean sure, other bots can possibly react to the poll. But as long as we filter out this bot, then we're fine.

```suggestion
      const reactionCount = reactionCount - 1;
      // excludes bot
      const reactionUsernames = await Promise.all(
        [...reactionUserIds.keys()]
          .filter((userId) => userId !== CLIENT_ID)
          .map(async (userId) => {
            const user = await interaction.client.users.fetch(userId);
            return user.username;
          })
      );
```